File: python_bot/handlers/start.py
from aiogram import Router, F
from aiogram.filters import CommandStart
from aiogram.types import Message

from keyboards.reply import get_lang_keyboard, get_main_menu
from utils.session import set_user_lang, get_user_lang
from utils.texts import TEXTS

router = Router()

@router.message(CommandStart())
async def cmd_start(message: Message):
    """
    /start buyrug'i kelganda ishlaydi.
    Til tanlash menyusini ko'rsatadi.
    """
    # Standart til xabari (barcha tillarda chiqarsh mumkin, hozircha matnlardan olyapmiz)
    text = TEXTS['uz']['choose_lang']
    
    await message.answer(
        text=text,
        reply_markup=get_lang_keyboard()
    )


@router.message(F.text.in_(["🇺🇿 O‘zbek tili", "🇷🇺 Rus tili", "🇬🇧 Ingliz tili"]))
async def language_selected(message: Message):
    """
    Foydalanuvchi tilni tanlaganda ishlaydi.
    """
    # Tanlangan tilga qarab xotiraga saqlaymiz
    if message.text == "🇺🇿 O‘zbek tili":
        lang = 'uz'
    elif message.text == "🇷🇺 Rus tili":
        lang = 'ru'
    elif message.text == "🇬🇧 Ingliz tili":
        lang = 'en'
    else:
        lang = 'uz'
        
    set_user_lang(message.from_user.id, lang)
    
    # Tanlangan til bo'yicha matnlarni olamiz
    texts = TEXTS[lang]
    
    # Foydalanuvchiga asosiy menyuni ko'rsatamiz
    await message.answer(
        text=texts['main_menu'],
        reply_markup=get_main_menu(lang)
    )

File: cf_bot/src/handlers/start.js
import { getLangKeyboard, getMainMenu } from '../keyboards/reply.js';
import { setUserLang } from '../utils/session.js';
import { TEXTS } from '../utils/texts.js';

export function setupStartHandlers(bot) {
    bot.start(async (ctx) => {
        const text = TEXTS['uz']['choose_lang'];
        await ctx.reply(text, getLangKeyboard());
    });

    const languages = ["🇺🇿 O‘zbek tili", "🇷🇺 Rus tili", "🇬🇧 Ingliz tili"];
    
    bot.hears(languages, async (ctx) => {
        const selectedText = ctx.message.text;
        let lang = 'uz';
        
        if (selectedText === "🇺🇿 O‘zbek tili") lang = 'uz';
        else if (selectedText === "🇷🇺 Rus tili") lang = 'ru';
        else if (selectedText === "🇬🇧 Ingliz tili") lang = 'en';

        setUserLang(ctx.from.id, lang);
        const texts = TEXTS[lang];

        await ctx.reply(texts.main_menu, getMainMenu(lang));
    });
}
